### PR TITLE
[Fixed]: Bash script 

### LINKED ISSUE #52: The `run.sh` script doesn't work for exercise that starts with a higher letter than `m`.
The run.sh gets 
```shell
spec_file="${spec_files[0]}"
```

The spec_files is made of 2 files, one called `<exercise>_spec.cr` and another one called `modified_test_spec.cr`.
Since that index can change based on what the <exercise> name starts on so will this make the script nonfunctional for a lot of exercises.

### PR DESCRIPTION
**Pull request by meatball133**
resolves #52

### CODE REVIEW

**Comment by meatball133** [comment]
To make this "fix" to the bash script I had to make an if statement to check if the file exists since the spec uses different file naming than the exercise.
Is file names something that I am allowed to change?
Otherwise, would these modifications I have done make it work for both, I want to say that I am no master at bash so it may not be a good script.

**Review by ErikSchierboom — COMMENTED** [review]
Could you maybe add a test case that would fail with the old code but passes the new one? This will guard against any regression errors.

> Is file names something that I am allowed to change?

You are!

**Comment by meatball133** [comment]
Well, the changes are more or less that I enable the code to work in both the new and old way.

I discussed in the issue what the issue was with index with file names.
The way normal "exercises" work is that the test files are named: <exercise_snake>_spec,

Although the way the spec file for the test generator doesn't work in the same way.
Since the exercise name which is given is not the same as the spec file.
The spec file has always the name of mini_bob_spec.
Meanwhile, the exercise given has a different name than <mini_bob>.

A solution would be to rename all the test files to have the same directory name as the spec file.
Otherwise so does this 2-way solution also work.

**Review by ErikSchierboom — APPROVED** [review]
Yes I understand the issue, but as this is something that is currently not working on you are implementing a fix, a test case is highly recommended. But your choice 🤷

**Comment by ErikSchierboom** [comment]
Well, you'd have to add a new directory here: https://github.com/exercism/crystal-test-runner/tree/main/tests with the (in)properly named files

**Comment by ErikSchierboom** [comment]
p.s. I'm happy for you to merge this and add a test case later BTW

**Comment by meatball133** [comment]
I think the test thing is good.
I am just thinking of how to do it since what the "normal" tests does is to compare an expected.json with a result.json.
Meaning it runs the code in docker with crystal on, and will thereby give that output.

The thing is that this file error is not raised by crystal nor the test-runner.
It is raised by the bash script, since it can't find the file.
That is the reason this is not a bug on the website.
What the bash script creates in then this happens is just an empty spec file.

Ohhh wait, you are right.
I could do code that would fail and then because there will be an empty test file, thereby it will succeed.

What do you about this approach?

**Comment by ErikSchierboom** [comment]
> The thing is that this file error is not raised by crystal nor the test-runner.
> It is raised by the bash script, since it can't find the file.

But the bash script _is_ used in the test runner: https://github.com/exercism/crystal-test-runner/blob/main/Dockerfile#L13

> I could do code that would fail and then because there will be an empty test file, thereby it will succeed.

Yeah, something like that sounds good

**Comment by meatball133** [comment]
> But the bash script is used in the test runner: https://github.com/exercism/crystal-test-runner/blob/main/Dockerfile#L13

Hmmm, could it be so that when you run the test runner online so is it just 1 file and that gets assigned and after that is a modified test file created. Meaning that this issue never appears since when the script were run there is only 1 file in the directory?

**Comment by meatball133** [comment]
Well that may be the proper solution to this problem, which is to remove that file when the script is finished.
What would you say about that?

**Comment by ErikSchierboom** [comment]
I think there might be some confusion here. Let's just merge this PR as is. It seems to be working fine.

File: bin/run.sh [hunk 1]
```diff
@@ -25,13 +25,8 @@ slug="$1"
...
 output_dir="${3%/}"
-if [[ -f "${input_dir}/spec/${snake_slug}_spec.cr" ]]
-then
-    spec_file="${input_dir}/spec/${snake_slug}_spec.cr"
-else
-    spec_files=( ${input_dir}/spec/*_spec.cr )
-    spec_file="${spec_files[0]}"
-fi
+config=$(jq -r '.files.test[0]' ${input_dir}/.meta/config.json)
+spec_file="${input_dir}/${config}"
```
Comment (ErikSchierboom) [inline bin/run.sh:29]: Could you check if this works?

```suggestion
spec_file="${input_dir}/$(jq -r '.files.test[0]' ${input_dir}/.meta/config.json)"
```

If not, you could maybe rename the variable a bit:

```suggestion
relative_spec_file=$(jq -r '.files.test[0]' ${input_dir}/.meta/config.json)
spec_file="${input_dir}/${relative_spec_file}"
```

**Pull request merged**